libraries/AP_InertialSensor/AP_InertialSensor_LSM9DS1.cpp

```diff
@@ -0,0 +1,573 @@
+
+    //Enable block data update, allow auto-increment during multiple byte read
+    _dev->write_register(LSM9DS1XG_CTRL_REG8, LSM9DS1XG_CTRL_REG8_BDU | LSM9DS1XG_CTRL_REG8_IF_ADD_INC);
+    hal.scheduler->delay_microseconds(1);
+
+    //Enable FIFO stream mode and set watermark at 32 samples
+    _dev->write_register(LSM9DS1XG_FIFO_CTRL, 0x1F | LSM9DS1XG_FIFO_CTRL_FMODE_FIFO);
+    hal.scheduler->delay_microseconds(1);
+
+    _dev->set_speed(AP_HAL::Device::SPEED_HIGH);
+
+    notify_accel_fifo_reset(_accel_instance);
+    notify_gyro_fifo_reset(_gyro_instance);
+}
+
+/*
+  start the sensor going
+ */
+void AP_InertialSensor_LSM9DS1::start(void)
+{
+    _gyro_instance = _imu.register_gyro(952, _dev->get_bus_id_devtype(DEVTYPE_GYR_LSM9DS1));
+    _accel_instance = _imu.register_accel(952, _dev->get_bus_id_devtype(DEVTYPE_ACC_LSM9DS1));
+
+    set_accel_orientation(_accel_instance, _rotation);
+    set_gyro_orientation(_gyro_instance, _rotation);
+
+
+    _set_accel_max_abs_offset(_accel_instance, 5.0f);
+
+    /* start the timer process to read samples */
+    _dev->register_periodic_callback(1000, FUNCTOR_BIND_MEMBER(&AP_InertialSensor_LSM9DS1::_poll_data, void));
+}
+
+
+uint8_t AP_InertialSensor_LSM9DS1::_register_read(uint8_t reg)
+{
+    uint8_t val = 0;
+
+    _dev->read_registers(reg, &val, 1);
+
+    return val;
+}
+
+
+void AP_InertialSensor_LSM9DS1::_register_write_xg(uint8_t reg, uint8_t val, bool checked)
+{
+    _dev->write_register(reg, val, checked);
+}
+
+
+void AP_InertialSensor_LSM9DS1::_gyro_init()
+{
+
+    _register_write_xg(LSM9DS1XG_CTRL_REG1_G, LSM9DS1XG_CTRL_REG1_G_ODR_G_952Hz |
+                                              LSM9DS1XG_CTRL_REG1_FS_G_2000DPS);
+    hal.scheduler->delay(1);
+
+    _register_write_xg(LSM9DS1XG_CTRL_REG4, LSM9DS1XG_CTRL_REG4_Zen_G |
+                                            LSM9DS1XG_CTRL_REG4_Yen_G |
+                                            LSM9DS1XG_CTRL_REG4_Xen_G);
+    _set_gyro_scale(G_SCALE_2000DPS);
+    hal.scheduler->delay(1);
+}
+
+void AP_InertialSensor_LSM9DS1::_accel_init()
+{
+
+    _register_write_xg(LSM9DS1XG_CTRL_REG6_XL, LSM9DS1XG_CTRL_REG6_XL_ODR_XL_952Hz |
+                                               LSM9DS1XG_CTRL_REG6_XL_FS_XL_16G);
+    hal.scheduler->delay(1);
+
+    _register_write_xg(LSM9DS1XG_CTRL_REG5_XL, LSM9DS1XG_CTRL_REG5_XL_Zen_XL |
+                                               LSM9DS1XG_CTRL_REG5_XL_Yen_XL |
+                                               LSM9DS1XG_CTRL_REG5_XL_Xen_XL);
+    _set_accel_scale(A_SCALE_16G);
+    hal.scheduler->delay(1);
+}
+
+void AP_InertialSensor_LSM9DS1::_set_gyro_scale(gyro_scale scale)
+{
+    /* scales values from datasheet in mdps/digit */
+    switch (scale) {
+    case G_SCALE_245DPS:
+        _gyro_scale = 8.75;
+        break;
+    case G_SCALE_500DPS:
+        _gyro_scale = 17.50;
+        break;
+    case G_SCALE_2000DPS:
+        _gyro_scale = 70;
```

Comment:
I don't think we will ever want a scale other than 2000dps, so we could just hardcode in the caller.